src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -580,22 +553,12 @@ private void AddServicesProducedInfo(IResource modelResource, IAnnotationHolder
 
                 sp.DcpServiceProducerAnnotation.Port = sp.ServiceBindingAnnotation.ContainerPort;
             }
-            else if (modelResource is ExecutableResource)
-            {
-                sp.DcpServiceProducerAnnotation.Port = sp.ServiceBindingAnnotation.Port;
-            }
-            else
+            else if (sp.ServiceBindingAnnotation.Port is not null && sp.ModelResource is not ProjectResource)
             {
-                if (sp.ServiceBindingAnnotation.Port is null)
-                {
-                    throw new InvalidOperationException($"The ServiceBindingAnnotation for resource {modelResourceName} must specify the Port");
-                }
+                // Project resources do not use ServiceBindingAnnotation port because they will have port auto-allocated by the orchestrator.
+                // The port specified by the ServiceBindingAnnotation will be applied to the Service objects and injected into clients instead.
```

Comment:
I don't want this to be project specific behavior.

Reply:
Makes sense. I made the change (and it makes sense. to me). 

The caveat is, if you want to take advantage of the service info injection into clients, you have to do it "manually" right now, using DCP `portForServing` template function, because we do not have standardized way of injecting that information into Executables like we have for Projects (there is no `ASPNETCORE_URLS` in Executable world). For example:

```csharp
builder.AddExecutable(
        "simple_node",
        "npx",
        @"C:\Users\karolz\code\simple-node",
        new string[] { "ts-node", "index.ts" }
    ).WithEnvironment("PORT", """{{- portForServing "simple_node" -}}""")
    .WithServiceBinding(7789, "http", "simple_node");
```

(in the example above the `PORT` environment variable will be filled with auto-allocated port).

Also, this can/will become simpler once we complete the "headless services" work item.